### PR TITLE
Helm: add missing s3 encryption parameters

### PR DESCRIPTION
**Pull request by rafaelpissolatto**
**What this PR does / why we need it**:
This MR adds and fixes missing encryption parameters of AWS S3 (KMS)

Example:

_values.yaml (HELM)_
```yaml
  storage:
    bucketNames:
      chunks: XXXXXX-LOKI-S3-BUCKET/chunks
      ruler: XXXXXX-LOKI-S3-BUCKET/ruler
      admin: XXXXXX-LOKI-S3-BUCKET/admin
    type: s3
    s3:
      s3: "s3://us-west-2"
      region: us-west-2
      insecure: false
      s3ForcePathStyle: false
      signatureVersion: v4
      sseEncryption: true
      sse:
        type: SSE-KMS
        kmsKeyId: xxxxx-xxxx-xxxx-xxxx-xxxxxx
```         

_config (loki)_
```yaml
      ...
      signature_version: v4
      sse_encryption: true
      sse:
        type: SSE-KMS
        kms_key_id: xxxxx-xxxx-xxxx-xxxx-xxxxxx 
```

(Also, based on this reference: https://grafana.com/docs/loki/latest/configuration/#aws_storage_config)

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Comment by JStickler** [comment]
Since this is adding missing parameters, they also need to be added to the documentation.  Please update https://github.com/grafana/loki/blob/main/production/helm/loki/reference.md.gotmpl.

**Comment by MichelHollands** [comment]
@rafaelpissolatto For Helm chart changes bump the Helm chart version in production/helm/loki/Chart.yaml and update production/helm/loki/CHANGELOG.md and production/helm/loki/README.md

File: docs/sources/setup/install/helm/reference.md [hunk 1]
```diff
@@ -2149,7 +2151,12 @@ null
     "s3": null,
     "s3ForcePathStyle": false,
     "secretAccessKey": null,
-    "signatureVersion": null
+    "signatureVersion": null,
```
Comment (MichelHollands) [inline docs/sources/setup/install/helm/reference.md:2154]: Can you run `make -C docs sources/setup/install/helm/reference.md` as mentioned in https://drone.grafana.net/grafana/loki/26714/5/2?

File: production/helm/loki/templates/_helpers.tpl [hunk 2]
```diff
@@ -228,6 +228,21 @@ s3:
   {{- with .signatureVersion }}
   signature_version: {{ . }}
   {{- end }}
+  {{- with .sseEncryption }}
+  sse_encryption: {{ . }}
+  {{- end }}
+  {{- with .sse }}
+  sse:
+    {{- with .type }}
```
Comment (MichelHollands) [inline production/helm/loki/templates/_helpers.tpl:236]: There is no type parameter in the default values. Shouldn't that be added?

**Pull request closed without merging**